cheroot/server.py

```diff
@@ -1600,6 +1603,7 @@ def bind(self, family, type, proto=0):
                 pass
 
         self.socket.bind(self.bind_addr)
+        self.bind_addr = self.socket.getsockname()[:2]  # TODO: keep separate
```

Comment:
This change seems dangerous to me. You're overwriting the state on the object... so where one could previously access `self.bind_addr` to determine what address was attempted to be bound, now one gets instead the addr than was bound. Plus, there's a TODO - not sure why not just do:

```
    @property
    def bound_addr(self):
        return self.socket.getsockname()[:2]
```

or even better, just let the caller call server.socket.getsockname() directly.

Reply:
Yeah, it's just a dirty hack I did while refactoring tests. I wanted to postpone its resolution and to not dig into it while focusing on tests rewrite.
I don't want to mix this into the current PR, so I've logged this in #70